### PR TITLE
feat: Introduce Multi Attribute Search for Publisher

### PR DESCRIPTION
**Pull request by menuka94**
### Proposed changes in this pull request
- Introduce (not fully implemented) Multi-attribute search for Publisher (No DAO changes)

### Follow up actions
- Research possible designs in database layer to support queries on aforementioned functionality
- Implement DAO methods to support Multi-attribute search

#### General

- [x] **Is this PR explained thoroughly?** All code changes must be accounted for in the PR description.
- [x] **Is the PR labeled correctly?**

#### Code

- [x] **Do you fully understand the introduced changes to the code?** If not ask for clarification, it might uncover ways to solve a problem in a more elegant and efficient way.
- [ ] **Does the PR introduce any inefficient database requests?** Use the debug server to check for duplicate requests.
- [ ] **Are all necessary strings marked for translation?** All strings that are exposed to users via the UI must be [marked for translation](https://docs.djangoproject.com/en/1.10/topics/i18n/translation/).

#### Security

- [x] **Confirm this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets.**

### CODE REVIEW

File: components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/impl/APIPublisherImpl.java [hunk 1]
```diff
@@ -1272,14 +1275,41 @@ public void deleteAPI(String identifier) throws APIManagementException {
...
+
+                String[] attributes = query.split(ATTRIBUTE_DELIMITER);
+                Map<String, String> attributeMap = new HashMap<>();
+
+                boolean isFullTextSearch = false;
+                String searchAttribute, searchValue;
+                if (!query.contains(KEY_VALUE_DELIMITER)) {
+                    isFullTextSearch = true;
+                } else {
+                    log.info("Search query: " + query);
```
Comment (uvindra) [inline components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/impl/APIPublisherImpl.java]: This should be log.debug

File: components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/impl/APIPublisherImpl.java [hunk 2]
```diff
@@ -1272,14 +1275,41 @@ public void deleteAPI(String identifier) throws APIManagementException {
...
+                    for (String attribute : attributes) {
+                        searchAttribute = attribute.split(KEY_VALUE_DELIMITER)[0];
+                        searchValue = attribute.split(KEY_VALUE_DELIMITER)[1];
+                        log.info(searchAttribute + KEY_VALUE_DELIMITER + searchValue);
```
Comment (uvindra) [inline components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/impl/APIPublisherImpl.java]: This should be log.debug

File: components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/impl/APIPublisherImpl.java [hunk 3]
```diff
@@ -1272,14 +1275,41 @@ public void deleteAPI(String identifier) throws APIManagementException {
...
+                        attributeMap.put(searchAttribute, searchValue);
+                    }
+                }
+
+                if (isFullTextSearch) {
+                    apiResults = getApiDAO().searchAPIs(roles, user, query, offset, limit);
+                } else {
+                    log.info("Attributes:", attributeMap.toString());
```
Comment (uvindra) [inline components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/impl/APIPublisherImpl.java]: This should be log.debug

**Comment by menuka94** [comment]
@uvindra Done!

**Pull request merged**